pkg/virtctl/templates/templates.go

```diff
@@ -39,3 +46,15 @@ func OptionsUsageTemplate() string {
 {{.InheritedFlags.FlagUsages | trimTrailingWhitespaces}}{{end}}
 `
 }
+
+// ExactArgs validate the number of input parameters
+func ExactArgs(nameOfCommand string, n int) cobra.PositionalArgs {
+	return func(cmd *cobra.Command, args []string) error {
+		if len(args) != n {
+			fmt.Printf("fatal: Number of input parameters is incorrect, %s accepts %d arg(s), received %d\n\n", nameOfCommand, n, len(args))
```

Comment:
From what I can tell, the incorrect number of args cause the command to fail today without this change. 

functionally, is the big benefit here that the user receives a more exact log message or is there something else going on here as well I'm missing?

Reply:
The Issue is that the cobra package error message is not really user friendly.  
it was "accepts %d arg(s), received %d".
The Idea is to first let the user get a better printer message and also to provide help in case it of such error.
it's very similar to how "git" command is working that in case the parameters are in correct, you also get help menu.

Reply:
yep, makes sense to me.